core/state/statedb.go

```diff
@@ -320,6 +341,7 @@ func (self *StateDB) Copy() *StateDB {
 	// ignore error - we assume state-to-be-copied always exists
 	state, _ := New(common.Hash{}, self.db)
 	state.trie = self.trie
+	state.all = self.all
```

Comment:
Doesn't this set them to be the same cache? So calls that get reverted will have the cache from the later transaction? I'm pretty sure the copied state should have a new, empty cache.